data/moves.ts

```diff
@@ -14542,14 +14542,16 @@ export const Moves: import('../sim/dex-moves').MoveDataTable = {
 			for (i in target.boosts) {
 				source.boosts[i] = target.boosts[i];
 			}
+
 			const volatilesToCopy = ['dragoncheer', 'focusenergy', 'gmaxchistrike', 'laserfocus'];
+			// we need to remove all crit stage volatiles first; otherwise copying e.g. dragoncheer onto a mon with focusenergy
+			// will crash the server (since addVolatile fails due to overlap, leaving the source mon with no hasDragonType to set)
+			for (const volatile of volatilesToCopy) source.removeVolatile(volatile);
 			for (const volatile of volatilesToCopy) {
```

Comment:
```suggestion
			for (const volatile of volatilesToCopy) {
				source.removeVolatile(volatile);
```
wouldn't this functionally do the same thing with one less iteration?

Reply:
Also, is this a concern for Costar?

Reply:
The point is that you need to remove the Focus Energy volatile in order to be able to properly copy the Dragon Cheer volatile and vice versa. Costar can presumably activate after having volatiles though use of Neutralising Gas or other methods.

Reply:
(I guess the other approach would be to avoid having codependent volatiles altogether by putting all of the logic in the Psych Up volatile which would have to work out whether it had been set by Dragon Cheer on a non-Dragon type in which case it would only add 1 to the crit ratio.)

Reply:
We can't move the `removeVolatile` call inside the same loop that tries to add volatiles since that would not remove `focusenergy` before `dragoncheer` is added; in this case fusing the loop changes the behavior. We could re-order the `volatilesToCopy` list, but I'm not sure (or tbh willing to exhaustively test) that would be robust, and it definitely wouldn't be robust against possible future additions or list re-orderings.

Reply:
What is the cartridge behavior of Costar after neutralizing gas?
e.g. roaring moon has 2 dd's and dragoncheer effect, flamigo switches in and copies all three boosts (+2 atk, +2 spe, +2 crit).
opponent switches weezing-galar (neutralizing gas) in
roaring moon switches out to hydreigon
weezing galar switches out

should flamigo after costar re-ups have no boosts? volatiles only?

Reply:
> What is the cartridge behavior of Costar after neutralizing gas? e.g. roaring moon has 2 dd's and dragoncheer effect, flamigo switches in and copies all three boosts (+2 atk, +2 spe, +2 crit). opponent switches weezing-galar (neutralizing gas) in roaring moon switches out to hydreigon weezing galar switches out
> 
> should flamigo after costar re-ups have no boosts? volatiles only?

It should have attempted to copy Hydreigon's stats, which in this case would be +0 Atk and +0 Spe. It also would copy Hydreigon's lack of crit volatile, so it should have a +0 crit rate again after Costar activates. It's identical to how Psych Up behaves.

Reply:
Great, that's the current behavior I have tested, then.